uxarray/_ugrid.py

```diff
@@ -10,12 +10,54 @@ def _read_ugrid(filepath):
     Returns: the xarray Dataset loaded during init.
     """
 
+    # TODO: obtain and change to Mesh2 construct, see Issue #27
     # simply return the xarray object loaded
-    return xr.open_dataset(filepath, mask_and_scale=False)
+
+    xr_ds = xr.open_dataset(filepath, mask_and_scale=False)
+
+    #
+    base_dv_mt = list(xr_ds.filter_by_attrs(cf_role="mesh_topology").keys())[0]
+
+    var_names_dict["Mesh2"] = base_dv_mt
+
+    coord_names = xr_ds[base_dv_mt].node_coordinates.split()
+
+    if len(coord_names) == 1:
+        var_names_dict["Mesh2_node_x"] = coord_names[0]
+    elif len(coord_names) == 2:
+        var_names_dict["Mesh2_node_x"] = coord_names[0]
+        var_names_dict["Mesh2_node_y"] = coord_names[1]
+    elif len(coord_names) == 3:
+        var_names_dict["Mesh2_node_x"] = coord_names[0]
+        var_names_dict["Mesh2_node_y"] = coord_names[1]
+        var_names_dict["Mesh2_node_z"] = coord_names[2]
+
+    # set #nodes use x coordinates, y or z will be the same and can also be used
+    coord_dim_name = xr_ds[var_names_dict["Mesh2_node_x"]].dims
+    var_names_dict["nMesh2_node"] = coord_dim_name[0]
+
+    face_node_names = xr_ds[base_dv_mt].face_node_connectivity.split()
+
+    face_node_name = face_node_names[0]
+    var_names_dict["Mesh2_face_nodes"] = xr_ds[face_node_name].name
+    var_names_dict["nMesh2_face"] = xr_ds[face_node_name].dims[0]
+    var_names_dict["nMaxMesh2_face_nodes"] = xr_ds[face_node_name].dims[1]
+
+    if len(coord_names) == 2:
+        # set coordinates
+        xr_ds = xr_ds.set_coords(
+            [var_names_dict["Mesh2_node_x"], var_names_dict["Mesh2_node_y"]])
+    else:
+        xr_ds = xr_ds.set_coords([
+            var_names_dict["Mesh2_node_x"], var_names_dict["Mesh2_node_y"],
+            var_names_dict["Mesh2_node_z"]
+        ])
+
+    return xr_ds
```

Comment:
`var_names_dict` gets updated after leaving this ffunction, but for the sake of code readability, should we also return that in here:

`return xr_ds, var_names_dict`

and change Line 155 in grid.py as follows:

`self.ds, self.var_names_dict = _read_ugrid(self.filepath, self.var_names_dict)`?

I am not  sure which way to go though.

Reply:
makes sense. fixed.